src/app/ecosystem-explorer/utils/getCategoriesFromDirectory.ts

```diff
@@ -0,0 +1,17 @@
+import { z } from 'zod'
+
+import { ECOSYSTEM_CATEGORIES_DIRECTORY_PATH } from '@/constants/paths'
+
+import { readAndValidateYamlFiles } from '@/utils/yamlUtils'
+
+import { DirectoryCategorySchema } from '../schemas/CategorySchemas'
+
+export function getCategoriesFromDirectory() {
+  const rawCategories = readAndValidateYamlFiles(
+    ECOSYSTEM_CATEGORIES_DIRECTORY_PATH,
+  )
+
+  const categories = z.array(DirectoryCategorySchema).parse(rawCategories)
+
+  return categories
+}
```

Comment:
Could `getCategoriesFromDirectory` and `getSubcategoriesFromDirectory` be a single function? There are quite similar, but happy to keep it this way if more readable.

Reply:
I initially combined them in one function, then split that function into two so they each had a single responsibility.

These are single-use functions, so to speak, so maybe one function is fine, yes.

Reply:
Yes - Let's have a single function.

Reply:
✅